packages/graph/hash_graph/lib/graph/src/api/web/data_type.rs

```diff
@@ -0,0 +1,145 @@
+//! Web routes for Data Types CRU operations.
+
+use axum::{
+    extract::Path,
+    http::StatusCode,
+    response::IntoResponse,
+    routing::{get, post},
+    Extension, Json, Router,
+};
+use serde::{Deserialize, Serialize};
+use utoipa::{Component, OpenApi};
+use uuid::Uuid;
+
+use super::{
+    api_resource::RoutedResource,
+    error::{modify_report_to_status_code, query_report_to_status_code},
+};
+use crate::{
+    datastore::Datastore,
+    types::{schema::DataType, AccountId, BaseId, Qualified, QualifiedDataType, VersionId},
+};
+
+#[derive(OpenApi)]
+#[openapi(
+    handlers(
+        create_data_type,
+        get_data_type,
+        // get_data_type_many,
+        update_data_type
+    ),
+    components(CreateDataTypeRequest, UpdateDataTypeRequest, AccountId, BaseId, QualifiedDataType),
+    tags(
+        (name = "DataType", description = "Data Type management API")
+    )
+)]
+pub struct DataTypeResource;
+
+impl RoutedResource for DataTypeResource {
+    /// Create routes for interacting with data types.
+    fn routes<D: Datastore>() -> Router {
+        // TODO: The URL format here is preliminary and will have to change.
+        Router::new().nest(
+            "/data-type",
+            Router::new()
+                .route("/", post(create_data_type::<D>).put(update_data_type::<D>))
+                // .route("/query", get(get_data_type_many))
+                .route("/:version_id", get(get_data_type::<D>)),
+        )
+    }
+}
+
+#[derive(Serialize, Deserialize, Component)]
+struct CreateDataTypeRequest {
+    #[component(value_type = Any)]
+    schema: DataType,
+    account_id: AccountId,
+}
+
+#[utoipa::path(
+    post,
+    path = "/data-type",
+    request_body = CreateDataTypeRequest,
+    tag = "DataType",
+    responses(
+      (status = 201, content_type = "application/json", description = "Data type created successfully", body = QualifiedDataType),
```

Comment:
This also exposes the internal id. Is this intentional?

Reply:
This is intentional for now, as we're querying by version ID in the routes anyway. When/if we get rid of qualified, we'll have to revisit this.